modules/data.atmosphere/R/download.NOAA.R

```diff
@@ -0,0 +1,387 @@
+           cbind(noaa_data[[i]], rnoaa::gefs(noaa_var_names[i], lat.in, lon.in, raw=TRUE, time_idx = 1, forecast_time = "0600", date=gefs_date)$data)
+        }, error=function(e){
+          continue=FALSE
+          last_data_set_at = "0000"
+          PEcAn.logger::logger.info(paste0("Last data avaliable for ", Sys.Date(), " is the 12:00 midnight data point (0:00)."))
+          noaa_data
+          })
+      }
+      
+      if (hour > 12 && continue) {
+        noaa_data[[i]] = tryCatch({
+          last_data_set_at = "1200"
+          cbind(noaa_data[[i]], rnoaa::gefs(noaa_var_names[i], lat.in, lon.in, raw=TRUE, time_idx = 1, forecast_time = "1200", date=gefs_date)$data)
+        }, error=function(e){
+          continue=FALSE
+          last_data_set_at = "0600"
+          PEcAn.logger::logger.info(paste0("Last data avaliable for ", Sys.Date(), " is the 6:00 a.m. data point."))
+          noaa_data
+          })
+      }
+      
+      if (hour > 18 && continue) {
+        noaa_data[[i]] = tryCatch({
+          last_data_set_at = "1800"
+          cbind(noaa_data[[i]], rnoaa::gefs(noaa_var_names[i], lat.in, lon.in, raw=TRUE, time_idx = 1, forecast_time = "1800", date=gefs_date)$data)
+        }, error=function(e){
+          continue=FALSE
+          last_data_set_at = "1200"
+          PEcAn.logger::logger.info(paste0("Last data avaliable for ", Sys.Date(), " is the 12:00 noon data point."))
+          noaa_data
+          })
+      }
+    }
+  }
+  
+  if (verbose && include_forecast) {
+    PEcAn.logger::logger.info("Downloading forecast data...")
+  }
+  
+  #This loop attaches 16 day forecast data onto the data that currently exists.
+  if (include_forecast) {
+    for (i in 1:length(cf_var_names)) {
+      noaa_data[[i]] = cbind(noaa_data[[i]], rnoaa::gefs(noaa_var_names[i], lat.in, lon.in, raw=TRUE, time_idx = 2:64, forecast_time = last_data_set_at, date=gefs_date)$data)
+    }
+  }
+  
+  ###################################################
+  # Not all NOAA data units match the cf data standard.  In this next section, data are processed to
+  # confirm with the standard when necessary.
+  # The following is a list of variables which need to be processed:
+  # 1. NOAA's relative humidity must be converted to specific humidity
+  # 2. NOAA's measure of precipitation is the accumulation over 6 hours; cf's standard is precipitation per second
+  
+  if (verbose) {
+    PEcAn.logger::logger.info("Converting units.")
+  }
+  
+  #Convert NOAA's relative humidity to specific humidity
+  humid_index = cf_var_names == "specific_humidity"
+  
+  humid_data = noaa_data[humid_index][[1]]
+  temperature_data = noaa_data[cf_var_names == "air_temperature"][[1]]
+  pressure_data = noaa_data[cf_var_names == "air_pressure"][[1]]
+   
+  #Depending on the volume and dimensions of data you download, sometimes R stores it as a vector and sometimes
+  #as a matrix; the different cases must be processed with different loops.
```

Comment:
I'm not sure why this would be the case -- aren't we always grabbing all of the ensemble members by default.